Classes/YLRefreshHeaderView/YLRefreshHeaderView.m

```diff
@@ -31,26 +35,37 @@ - (void)setRefreshState:(YLRefreshHeaderViewState)refreshState {
 }
 
 - (void)setRefreshState:(YLRefreshHeaderViewState)refreshState animated:(BOOL)animated {
-  _refreshState = refreshState;
+  if (_refreshState != refreshState) {
+    _refreshState = refreshState;
 
-  void (^animationBlock)() = ^{
-    UIScrollView *strongScrollView = self.scrollView;
-    UIEdgeInsets contentInset = strongScrollView.contentInset;
-    // TODO(mglidden): make this support view controllers that have UIRectEdgeTop
-    contentInset.top = (refreshState == YLRefreshHeaderViewStateRefreshing) ? self.pullAmountToRefresh : 0;
-    strongScrollView.contentInset = contentInset;
-  };
-  void (^completionBlock)(BOOL) = ^(BOOL finished) {
-    if (self.refreshState == YLRefreshHeaderViewStateClosing) {
-      self.refreshState = YLRefreshHeaderViewStateNormal;
+    void (^animationBlock)() = ^{
+      UIScrollView *strongScrollView = self.scrollView;
+      UIEdgeInsets contentInset = strongScrollView.contentInset;
+      switch (refreshState) {
+        case YLRefreshHeaderViewStateRefreshing:
+          self.previousTopInset = contentInset.top;
+          contentInset.top = self.pullAmountToRefresh;
```

Comment:
Oh wait, I see what you mean now. This is a bit related to the previous comment about a positive contentInset.top. I think we can account for this case